### ISSUE TITLE
Support for long lines

### ISSUE THREAD

**Issue by schiegl** [vscode-neovim/vscode-neovim#573]
The extension ceases to work correctly on longer lines (most of the time). For example motion keys such as `w` don't work after a certain number of characters. Or highlighting a paragraph with `vip` doesn't highlight the entire paragraph.

This is rarely a problem in regular programming but often a problem when writing prose e.g. in latex or markdown. I know that I could hard-wrap the text (with the rewrap extension), but hard-wrapping doesn't work as well as soft-wrapping for me. It often doesn't work when editing or inserting text into an already existing paragraph. I always end up having to do it manually.

What is it that prevents the extension from working with longer lines? Can I increase a parameter to make it work in latex/markdown documents? I tried increasing the "Neovim Width" but that didn't help.

**Comment by QAston** [comment]
You can change this setting: `"vscode-neovim.neovimWidth": 1000,` to enable longer lines

**Comment by schiegl** [comment]
As mentioned in the original post. Changing this setting does not solve the problem. Increasing the width from 1000 to 5000 does not let me access/modify longer lines than before.

**Comment by asvetliakov** [comment]
fyi you need to restart vscode after changing this setting

Try vim command `set wrap`. The ext is explicitly make it `nowrap`, but the reasons for it might be not relevant anymore. The visual mode highlight likely would be wrong though, but it's not fixable

**Comment by schiegl** [comment]
> fyi you need to restart vscode after changing this setting

I restarted vscode but it did not help.

> Try vim command set wrap. The ext is explicitly make it nowrap, but the reasons for it might be not relevant anymore. The visual mode highlight likely would be wrong though, but it's not fixable

Adding `set wrap` to my `init.vim` didn't change anything. But setting it manually (after starting) did.

However, the behavior is still buggy. Sometimes I can navigate with `w` and `b` around the whole line, sometimes it won't let me go further. Same goes for `A`, `gg` and `G`. This seems to happen more when I modified the line e.g. removed/added a word. It seems to lose track of how long the line actually is (with `A`) and what the "first" soft-wrapped line is (with `gg`).
